Sources/SwiftProtobuf/TextFormatEncodingVisitor.swift

```diff
@@ -59,6 +59,24 @@ internal struct TextFormatEncodingVisitor: Visitor {
     self.encoder = encoder
   }
 
+  private mutating func startField(lookingUp fieldNumber: Int) throws {
+      // This compact form of startField(lookingUp:) slows the entire
+      // TextFormat encoding process by over 2x.
+      //let buff = try protoFieldName(for: fieldNumber)
+      //startField(name: buff)
+
+      // This manually unrolled version is *much* faster.
+      if let protoName = nameMap?.names(for: fieldNumber)?.proto {
+          encoder.startField(name: protoName.utf8Buffer, inExtension: inExtension)
+      } else if let protoName = nameResolver[fieldNumber] {
+          encoder.startField(name: protoName, inExtension: inExtension)
+      } else if let extensionName = extensions?[fieldNumber]?.protobufExtension.fieldName {
+          encoder.startField(name: extensionName, inExtension: inExtension)
+      } else {
+          throw TextFormatEncodingError.missingFieldNames
```

Comment:
💡!

Even apart from unknown fields, I just realized that text serialization can always succeed even without field names since we can always fall back to the number form.  I'll refactor this a bit to make that work.